src/richie/apps/core/templates/menu/header_menu.html

```diff
@@ -2,43 +2,38 @@
-            {% if child.selected or child.attr.redirect_url == current_page.get_absolute_url %} topbar__item--selected{% endif %}
-            {% if child.ancestor %} topbar__item--ancestor{% endif %}
-            {% if child.sibling %} topbar__item--sibling{% endif %}
-            {% if child.descendant %} topbar__item--descendant{% endif %}"
-        >
-            {% comment %}Dropdown menu for children are only for page with index page
-            extension with a specific option enabled{% endcomment %}
-            {% if menu_options.allow_submenu and child.children %}
-                    <button 
-                      aria-expanded="false"
-                      aria-haspopup="true"
-                      aria-controls="dropdown-{{ child.id }}"
-                      id="dropdown-button-{{ child.id }}"
-                    >
-                      {{ child.get_menu_title }}
-                      <svg class="icon" aria-hidden="true">
-                        <use xlink:href="#icon-chevron-down"></use>
-                      </svg>
-                    </button>
-                    <nav 
-                      class="topbar__sublist"
-                      id="dropdown-{{ child.id }}"
-                      aria-labelledby="dropdown-button-{{ child.id }}"
-                    >
-                      <ul role="menu">
-                        {% show_menu from_level to_level extra_inactive extra_active template "" "" child %}
-                      </ul>
-                    </nav>
-            {% else %}
-                <a href="{{ child.attr.redirect_url|default:child.get_absolute_url }}">
-                    {{ child.get_menu_title }}
-                </a>
-            {% endif %}
-        </li>
+        {% with page_url=current_page.get_absolute_url redirect_url=child.attr.redirect_url %}
+            <li class="topbar__item
+                {% if menu_options.allow_submenu and child.children %}dropdown{% endif %}
+                {% if menu_options.allow_submenu and menu_options.menu_color %} topbar__item--{{ menu_options.menu_color }}{% endif %}
+                {% if child.ancestor %} topbar__item--ancestor{% endif %}
+                {% if child.sibling %} topbar__item--sibling{% endif %}
+                {% if child.descendant %} topbar__item--descendant{% endif %}
+                {% if child.selected %} topbar__item--selected
+                {% elif redirect_url and page_url == redirect_url %} topbar__item--selected
+                {% endif %}"
+            >
+                {% comment %}Dropdown menu for children are only for page with index page
+                extension with a specific option enabled{% endcomment %}
+                {% if menu_options.allow_submenu and child.children %}
+                        <button aria-expanded="false" aria-haspopup="true" aria-controls="dropdown-{{ child.id }}" id="dropdown-button-{{ child.id }}">
```

Comment:
FMI, why removing line breaks here ?

Reply:
Just my IDE that reformat the file after I changed indentation and the readability was poor so I put this tag inline but indeed I could revert this.

Reply:
No need to revert :+1:

Reply:
Too late 😄